### PR TITLE
test(graphql-relational-transformer): add some more unit tests for has-many references ddb

### PR DESCRIPTION
**Pull request by atierian**
#### Description of changes
Adds some more unit tests for has-many references based relationships with DDB data sources.

##### CDK / CloudFormation Parameters Changed
None

#### Issue #, if available
N/A

#### Description of how you validated changes
Unit tests passing.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] ~Relevant documentation is changed or added (and PR referenced)~
- [ ] ~New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies~
- [ ] ~Any CDK or CloudFormation parameter changes are called out explicitly~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts [hunk 1]
```diff
@@ -49,6 +49,205 @@ test('has many query', () => {
...
+    transformers: [
+      new ModelTransformer(),
+      new PrimaryKeyTransformer(),
+      new HasOneTransformer(),
+      new HasManyTransformer(),
+      new BelongsToTransformer(),
+    ],
+    dataSourceStrategies: {
+      Member: DDB_DEFAULT_DATASOURCE_STRATEGY,
+      Project: DDB_DEFAULT_DATASOURCE_STRATEGY,
```
Comment (AaronZyLee) [inline packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts:233]: It is better to have the amplify table strategy as well in this test case as it is a common case in Gen2

**Review by AaronZyLee — APPROVED** [review]

File: packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts [hunk 2]
```diff
@@ -49,6 +49,205 @@ test('has many query', () => {
...
+      teamID: ID! @index(name: "foo")
+      team: Team @belongsTo(references: ["teamID"])
+    }`;
+
+  expect(() =>
+    testTransform({
+      schema: inputSchema,
+      transformers: [new ModelTransformer(), new IndexTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+    }),
+  ).toThrowError('Invalid @hasMany directive on Team.members - indexName is not supported with DDB references.');
```
Comment (phani-srikar) [inline packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts:70]: I've got one naive question about how do we determine the GSI to use if no `indexName` is supported?

File: packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts [hunk 3]
```diff
@@ -49,6 +49,205 @@ test('has many query', () => {
...
+    }`;
+
+  // TODO: This should fail -- find appropriate place to validate.
+  // expect(() =>
+  //   testTransform({
+  //     schema: inputSchema,
+  //     transformers: [new ModelTransformer(), new HasManyTransformer(), new BelongsToTransformer()],
+  //   }),
+  // ).toThrowError();
+});
```
Comment (phani-srikar) [inline packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts:175]: nit: are there future work tickets to implement these validations?

File: packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts (hunk 2 above)
Comment (atierian) [reply packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts:70]: We create it ourselves with the format of `gsi-[TypeName].[FieldName]`. This aligns with what we are doing today when an index name isn't provided for `hasMany` / `belongsTo`.

File: packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts (hunk 3 above)
Comment (atierian) [reply packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-has-many-references-transformer.test.ts:175]: Just these TODOs and the [feature branch PRs description](https://github.com/aws-amplify/amplify-category-api/pull/2370) for larger things. There are quite a few of them in the feature branch already.

**Review by phani-srikar — APPROVED** [review]

**Pull request merged**